conda/cli/conda_argparse.py

```diff
@@ -117,9 +117,10 @@ def do_call(arguments: argparse.Namespace, parser: ArgumentParser):
     """
     # First, check if this is a plugin subcommand; if this attribute is present then it is
     if getattr(arguments, "plugin_subcommand", None):
-        _run_command_hooks("pre", arguments.plugin_subcommand.name, sys.argv[2:])
+        plugin_arguments = sys.argv[2:]
+        _run_command_hooks("pre", arguments.plugin_subcommand.name, plugin_arguments)
         result = arguments.plugin_subcommand.action(sys.argv[2:])
-        _run_command_hooks("post", arguments.plugin_subcommand.name, sys.argv[2:])
+        _run_command_hooks("post", arguments.plugin_subcommand.name, plugin_arguments)
```

Comment:
this means that the pre-command can modify the argument list and post-command will get the modified list, however the plugin itself will still receive the unmodified args

I can see value in allowing a pre-command plugin to modify the arguments before passing it to the subcommand but that is not what is currently happening and if this is a desired feature we should document it as such

Reply:
I don't think the pre command hook or the subcommand should be able to modify the arguments, we should handle `sys.argv` once, before all that and make it immutable and then pass them through the command flow. If we ever want to enable `sys.argv` modification it should become an own feature

Reply:
Perhaps it's better to make a copy and then pass it in as a tuple?